src/applications/disability-benefits/all-claims/utils/on-submit/purge-toxic-exposure-data.js

```diff
@@ -0,0 +1,238 @@
+import { isEmpty, pickBy, cloneDeep } from 'lodash';
+import {
+  ADDITIONAL_EXPOSURES,
+  GULF_WAR_1990_LOCATIONS,
+  GULF_WAR_2001_LOCATIONS,
+  HERBICIDE_LOCATIONS,
+} from '../../constants';
+
+/**
+ * Map of exposure types to their details keys and location constants.
+ * Used to dynamically process all toxic exposure types.
+ * This is the source of truth for all toxic exposure data structure.
+ *
+ * @constant {Object}
+ */
+export const EXPOSURE_TYPE_MAPPING = {
+  gulfWar1990: {
+    detailsKey: 'gulfWar1990Details',
+    locations: GULF_WAR_1990_LOCATIONS,
+  },
+  gulfWar2001: {
+    detailsKey: 'gulfWar2001Details',
+    locations: GULF_WAR_2001_LOCATIONS,
+  },
+  herbicide: {
+    detailsKey: 'herbicideDetails',
+    locations: HERBICIDE_LOCATIONS,
+    otherKey: 'otherHerbicideLocations',
+  },
+  otherExposures: {
+    detailsKey: 'otherExposuresDetails',
+    exposureTypes: ADDITIONAL_EXPOSURES,
+    otherKey: 'specifyOtherExposures',
+  },
+};
+
+/**
+ * Checks if any conditions other than 'none' are selected.
+ *
+ * @param {Object.<string, boolean>} conditions - Condition selections
+ * @returns {boolean} True if any non-'none' conditions are selected
+ */
+const hasSelectedConditions = conditions => {
+  if (!conditions) return false;
+  return Object.keys(conditions).some(
+    key => key !== 'none' && conditions[key] === true,
+  );
+};
+
+/**
+ * Helper function that processes a single exposure type, removing orphaned and
+ * non-applicable data. Called by purgeToxicExposureData for each exposure type
+ * (gulfWar1990, gulfWar2001, herbicide, otherExposures).
+ *
+ * Handles four main cleanup scenarios for each exposure type:
+ * 1. Removes null exposure fields as orphaned data
+ * 2. Removes orphaned details when main selection is missing
+ * 3. Removes entire section when all values are false/none or section is empty
+ * 4. Removes "other" fields when null, invalid, or user selected 'none'
+ *
+ * @param {Object} toxicExposure - Toxic exposure data object
+ * @param {Object.<string, boolean>} [toxicExposure[exposureType]] - Location/exposure selections as boolean values
+ * @param {Object} [toxicExposure[detailsKey]] - Detail objects (dates, descriptions) for selected locations
+ * @param {string} exposureType - Exposure type key (e.g., 'gulfWar1990')
+ * @param {Object} mapping - Mapping configuration for exposure type
+ * @param {string} mapping.detailsKey - Key for details object
+ * @param {string} [mapping.otherKey] - Key for other/specify fields
+ * @returns {Object} Toxic exposure object with orphaned/non-applicable data removed
+ */
+const purgeExposureDetails = (toxicExposure, exposureType, mapping) => {
+  const { detailsKey, otherKey } = mapping;
+  const result = { ...toxicExposure };
+
+  // Check conditions for various cleanup scenarios
```

Comment:
Nit for after release: might be helpful to have some examples in the docstrings showing what these steps are doing, since the data structure is kind of difficult to visualize from the code